bokeh/server/protocol/receiver.py

```diff
@@ -0,0 +1,118 @@
+''' Assemble websocket wire message fragments into complete Bokeh Server
+message objects that can be processed.
+
+'''
+from __future__ import absolute_import
+
+from six import b
+from tornado.concurrent import return_future
+
+from ..exceptions import MessageError, ProtocolError, ValidationError
+
+HMAC           = 0
+HEADER         = 1
+METADATA       = 2
+CONTENT        = 3
+BUFFER_HEADER  = 4
+BUFFER_PAYLOAD = 5
```

Comment:
A possibly-more-concise way to do this state machine is to make each state a function, and have a variable containing the current function. Like `self.current_consumer = consume_hmac` kind of thing. Then you avoid the "if" statements and the enum itself. Just a thought